spec/models/vm/retirement_management_spec.rb

```diff
@@ -55,22 +51,20 @@
 
   it "#retire_now with userid" do
     event_name = 'request_vm_retire'
-    event_hash = {:vm => @vm, :host => @vm.host, :type => "ManageIQ::Providers::Vmware::InfraManager::Vm",
-                  :retirement_initiator => "user", :userid => 'freddy'}
-    options = {:zone => @zone.name}
+    event_hash = {:userid => 'admin', :vm => @vm, :host => @vm.host, :type => "ManageIQ::Providers::Vmware::InfraManager::Vm"}
+    options = {:zone => @zone.name, :user_id => user.id, :group_id => MiqGroup.last.id, :tenant_id => Tenant.last.id}
 
     expect(MiqEvent).to receive(:raise_evm_event).with(@vm, event_name, event_hash, options).once
 
-    @vm.retire_now('freddy')
+    @vm.retire_now('admin')
```

Comment:
```suggestion
    @vm.retire_now(user.id)
```

may need to change event hash.